hire/hireapp/models.py

```diff
@@ -0,0 +1,288 @@
+
+from django.contrib.auth.models import User
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django.utils.text import slugify
+
+def name_from_attached_user(user):
+    name = ""
+    if user.first_name or user.last_name:
+        name = "%s %s" % (user.first_name, user.last_name)
+    else:
+        name = user.username
+    return name
+
+class Profile(models.Model):
+    STAFF = 1
+    COMPANY_CONTACT = 2
+    EMPLOYEE = 3
+    UNDEFINED_PROFILE = 4
+    PROFILE_CHOICES = (
+        (STAFF, "Staff"),
+        (COMPANY_CONTACT, "Company Contact"),
+        (EMPLOYEE, "Future Employee"),
+        (UNDEFINED_PROFILE, "Non-Classified Profile"),
+    )
+    user = models.OneToOneField(
+        User,
+        on_delete=models.RESTRICT)
+    phone_number = models.CharField(
+        max_length=12,
+        blank=True,
+        null=True,
+        unique=True
+    )
+    profile_type = models.PositiveSmallIntegerField(
+        choices=PROFILE_CHOICES,
+        null=True,
+        blank=True)
+
+
+    def __str__(self):
+        roles = dict(self.PROFILE_CHOICES)
+        if self.profile_type:
+            label = roles[self.profile_type]
+        else:
+            label = "Orphaned Profile"
+        return "%s: %s" % (label, self.user.username)
+
+@receiver(post_save, sender=User)
+def create_or_update_user_profile(sender, instance, created, **kwargs):
+    if created:
+        Profile.objects.create(user=instance)
+    try:
+        instance.profile.save()
+    except User.profile.RelatedObjectDoesNotExist:
+        Profile(user=instance).save()
+
+
+class Employee(models.Model):
+    # Intentionally leaving this a text field
+    # until we better understand hiring needs
+    user = models.OneToOneField(
+        User,
```

Comment:
This file is the code that relates to your data model. The class (e.g., `Employee`) represents the database table. The fields (e.g., `skills`) represents a field/column in that database.